src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -235,92 +73,173 @@ const bool Terminal::IsCopyOnSelectActive() const noexcept
 }
 
 // Method Description:
-// - Select the sequence between delimiters defined in Settings
+// - Perform a multi-click selection at viewportPos expanding according to the expansionMode
 // Arguments:
-// - position: the (x,y) coordinate on the visible viewport
-void Terminal::DoubleClickSelection(const COORD position)
+// - viewportPos: the (x,y) coordinate on the visible viewport
+// - expansionMode: the SelectionExpansionMode to dictate the boundaries of the selection anchors
+void Terminal::MultiClickSelection(const COORD viewportPos, SelectionExpansionMode expansionMode)
 {
-#pragma warning(suppress : 26496) // cpp core checks wants this const but .Clamp() can write it.
-    COORD positionWithOffsets = _ConvertToBufferCell(position);
-
-    // scan leftwards until delimiter is found and
-    // set selection anchor to one right of that spot
-    _selectionAnchor = _ExpandDoubleClickSelectionLeft(positionWithOffsets);
-    THROW_IF_FAILED(ShortSub(_selectionAnchor.Y, gsl::narrow<SHORT>(ViewStartIndex()), &_selectionAnchor.Y));
-    _selectionVerticalOffset = gsl::narrow<SHORT>(ViewStartIndex());
+    // set the selection pivot to expand the selection using SetSelectionEnd()
+    _selectionPivot = _ConvertToBufferCell(viewportPos);
+    _buffer->GetSize().Clamp(_selectionPivot);

```

Comment:
why does ConvertToBufferCell not clamp on its own? Will this even be necessary with ConverToBufferCell?